packages/addon-dev/src/rollup-public-entrypoints.ts

```diff
@@ -5,7 +5,14 @@ import minimatch from 'minimatch';
 import type { Plugin } from 'rollup';
 
 function normalizeFileExt(fileName: string) {
-  return fileName.replace(/\.ts|\.hbs|\.gts|\.gjs$/, '.js');
```

Comment:
Can you please check your use case and see if this implementation also solves it? 

```js
return fileName.replace(/(?<!\.d)\.ts|\.hbs|\.gts|\.gjs$/, '.js');
```

The idea here is "negative lookbehind assertions". 

If this works, we can stick with only a single regex replace instead of nesting them.

Reply:
Thank you, I checked that your solution works in https://github.com/ijlee2/embroider-css-modules/pull/46.